Modules/Brand/Http/Services/BrandService.php

```diff
@@ -0,0 +1,287 @@
+                return '<a href="'.url('/brand/details/' . $row->id).'"><span class="fas fa-eye"></span></a> <a href="'.url('/product/' . $row->id).'"><span class="fas fa-list"></span></a>  <a href="javascript:void(0)" onclick="event.preventDefault(); confirmDelete('.$row->id.');"><span class="fas fa-trash"></span></a> ';
+            })
+            ->rawColumns(['action'])
+            ->make(true);
+        $resultData = $result->getData(true);
+        $resultData['recordsFiltered'] = $filteredCount;
+        $resultData['draw'] = $request->draw;
+
+        return $resultData;
+    }
+
+    /**
+     * Details of brand
+     *
+     * @param $id
+     * @return mixed
+     */
+    public function detail($id): mixed
+    {
+
+        $brand = User::findOrFail($id);
+        $detail = $brand->profile;
+        $category = Category::where('id',$detail->prime_cat)->first();
+        $headQuarter = Country::where('id',$detail->headquatered)->first();
+        $productMade = Country::where('id',$detail->product_made)->first();
+        $productShipped = Country::where('id',$detail->product_shipped)->first();
+
+        $detail['name'] = $brand->first_name.' '.$brand->last_name;
+        $detail['email'] = $brand->email ?? '';
+        $detail['prime-category'] = $category->title ?? '';
+        $detail['head-quarter'] = $headQuarter->name ?? '';
+        $detail['product-made'] = $productMade->name ?? '';
+        $detail['product-shipped'] = $productShipped->name ?? '';
+
+        return $detail;
+    }
+
+    /**
+     * Delete Brand
+     *
+     * @param $id
+     * @return array
+     */
+    public function delete($id): array
+    {
+        $user = User::findOrFail($id);
+        $user->delete();
+        return [
+            'res' => true,
+            'msg' => 'Successfully deleted',
+            'data' => ''
+        ];
+    }
+
+    /**
+     * Get all country
+     *
+     * @return array
+     */
+    public function getCountry(): array
+    {
+        $countries = Country::orderBy('name', 'ASC')->get();
+        $data = array();
+        foreach ($countries as $country) {
+            $data[] = array(
+                'id' => $country->id,
+                'country_code' => $country->shortname,
+                'country_name' => $country->name,
+                'phone_code' => $country->phonecode
+            );
+        }
+
+        return $data;
+    }
+
+    /**
+     * Get a listing of only main product categories
+     *
+     * @return array
+     */
+    public function getCategory(): array
+    {
+        $data = [];
+        $mainCategories = Category::where('parent_id', 0)->where('status', '1')->orderBy('title', 'ASC')->get();
+        if ($mainCategories) {
+            foreach ($mainCategories as $mainCategory) {
+                $data[] = array(
+                    'category' => $mainCategory->title,
+                    'cat_id' => $mainCategory->id
+                );
+            }
+        }
```

Comment:
Update the query here also, you only need category title and id so pluck those only.

Reply:
Fixed